magpie-aws/src/main/java/io/openraven/magpie/plugins/aws/discovery/AWSUtils.java

```diff
@@ -200,7 +202,18 @@ public static GetMetricStatisticsResponse getCloudwatchMetricStatistics( String
       return client.getMetricStatistics(request);
     }
   }
+
   public static String getAwsAccountId() {
     return StsClient.create().getCallerIdentity().account();
   }
+
+  public static <BuilderT extends AwsClientBuilder<BuilderT, ClientT>, ClientT> ClientT
+  apply(AwsClientBuilder<BuilderT, ClientT> builder, Region region) {
+    // Remap magpie clients to local environment
+    if (System.getProperty("aws.local.endpoint") != null) {
+      builder.endpointOverride(URI.create(System.getProperty("aws.local.endpoint")));
+    }
+    builder.region(region);
+    return builder.build();
+  }
```

Comment:
This violates the rule of not having any test related code in running code.
Imo it's better to extract creating client before the given discover function like it's DynamoDbDiscovery. Then either production code or test code will create client needed for thier purpose without relying on checking env variable.

Reply:
I don't think this is necessarily test related code so much as a way to allow users to change their AWS endpoint. It also happens to work for the test environment.  

Can you link to a code example @PrzemyslawTusinski of what you're thinking?

Reply:
@PrzemyslawTusinski  I was also thinking about kind of Dependency Inversion for discover function, to provide prepared clients targeting for different purpose (PROD vs TESTs), but:
1. We are creating clients depends on Regions for each iteration over configured Regions for discovery (AWSDiscoveryPlugin:98). So i doesn't looks good to me to create all region dependent clients before we start
2. Also we think about running Magpie against the Demo stack to not reach the real AWS for any kind of simulation.
But if you see better approach here I would rather

Reply:
Since we are using this code not only in test this approach is fine

Reply:
Let's change the variable though, I don't like the `local` name and it also conflicts with how we assess other Magpie related variables.  It should be `MAGPIE_AWS_ENDPOINT`.

Reply:
But along with region we also configure endpointOverride property as well. Maybe `configure`. From that perspective `configure` name could be even more transparent about what is method about

Reply:
I *think* region will get picked up automatically from the `AWS_REGION` environmental variable.

Reply:
@kickroot  there we are setting region for client from config.yml. For each chosen region where user requested discovery.